### DISCUSSION TITLE
Can I publish a public package from a private repository?

### DISCUSSION THREAD

**Discussion by chrisfillmore** [community/community#29848] [Actions] [answered]
The documentation here
https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#overview-of-package-permissions-and-visibility

and here
https://docs.github.com/en/packages/learn-github-packages/publishing-a-package#about-published-packages

states

> You can publish packages in a public repository (public packages) to share with all of GitHub, or in a private repository (private packages) to share with collaborators or an organization.

But it doesn't say that you can't publish a public package from a private repository.

I published a package from a private repository. The package is private. But there is a button in package settings to change visibility to public:

![image](https://user-images.githubusercontent.com/7190265/185228016-8a3fa97a-e902-4edb-9147-1c20cc9e984a.png)

So, can I make it public? Or will this cause another problem? (maybe fail to install?)

**Comment by radar-blip** [comment, marked as answer]
Changing visibility to Public will do just that. It shouldn't cause any problems.

**Reply by chrisfillmore** [reply]
Ok on the basis of your confident answer, I'm going to change visibility to public and report back with the results!

**Reply by chrisfillmore** [reply]
Thank you for replying!

**Reply by radar-blip** [reply]
No problem. Can't wait for the results!

**Reply by radar-blip** [reply]
Well, what happened?? 😄

**Reply by chrisfillmore** [reply]
Seems to work just fine! The package is public!

**Reply by chrisfillmore** [reply]
Thanks again!

**Reply by radar-blip** [reply]
Awesome! Glad all went well. 
Have a great day!
